context/index.d.ts

```diff
@@ -0,0 +1,17 @@
+import type { ReadableAtom } from '../atom'
+
+// This isn't a part of the public API for now
+type Context = unknown
+
+export type ContextGetter = <T extends ReadableAtom>(store: T) => T
+export type WithCtx = {
```

Comment:
Too simple type to be exported. Let’s just use `& { ctx: ContextGetter }` in lifecycle